src/System.Data.OleDb/src/System/Data/Common/AdapterUtil.cs

```diff
@@ -0,0 +1,1531 @@
+            return InvalidCast(error, null);
+        }
+        static internal InvalidCastException InvalidCast(string error, Exception inner) {
+            InvalidCastException e = new InvalidCastException(error, inner);
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal InvalidOperationException InvalidOperation(string error) {
+            InvalidOperationException e = new InvalidOperationException(error);
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal TimeoutException TimeoutException(string error) {
+            TimeoutException e = new TimeoutException(error);
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal InvalidOperationException InvalidOperation(string error, Exception inner)
+        {
+            InvalidOperationException e = new InvalidOperationException(error, inner);
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal NotImplementedException NotImplemented(string error) {
+            NotImplementedException e = new NotImplementedException(error);
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal NotSupportedException NotSupported() {
+            NotSupportedException e = new NotSupportedException();
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal NotSupportedException NotSupported(string error) {
+            NotSupportedException e = new NotSupportedException(error);
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal InvalidCastException InvalidCast() {
+            InvalidCastException e = new InvalidCastException();
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal IOException IO(string error, Exception inner) {
+            IOException e = new IOException(error, inner);
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+        static internal InvalidOperationException DataAdapter(string error) {
+            return InvalidOperation(error);
+        }
+        static internal InvalidOperationException DataAdapter(string error, Exception inner) {
+            return InvalidOperation(error, inner);
+        }
+        static private InvalidOperationException Provider(string error) {
+            return InvalidOperation(error);
+        }
+        static internal ObjectDisposedException ObjectDisposed(object instance) {
+            ObjectDisposedException e = new ObjectDisposedException(instance.GetType().Name);
+            TraceExceptionAsReturnValue(e);
+            return e;
+        }
+
+        static internal ArgumentException InvalidPrefixSuffix() {
+            ArgumentException e = new ArgumentException(SR.GetString(SR.ADP_InvalidPrefixSuffix));
```

Comment:
Patterns like `SR.GetString(SR.ADP_InvalidPrefixSuffix)` can be regex twiddled to be jsut `SR.ADP_InvalidPrefixSuffix`